### PR TITLE
Add post_init configuration parameter on bootstrap

### LINKED ISSUE #291: Callback for custom operations after initial bootstrap
Our usecase involves creating a lot of database clusters that need to have some initial database setup (some specific databases with extensions) created on the cluster (so, only once after the initial initdb bootstrapping) before becoming usable.

So we would like to have a way for Patroni to invoke the running of these tasks after the db is created but before the node is deemed ready. 

I was thinking on adding a new `on_bootstrap` hook that gets executed after the initial bootstrap currently done [here](https://github.com/zalando/patroni/blob/master/patroni/postgresql.py#L973) and blocks until completion. 

Would this make sense?, can you think of any other way to achieve this?

### PR DESCRIPTION
**Pull request by zenitraM**
The script specified on it will be run after initdb, blocking cluster initialization until it finishes.

Fixes #291

### CODE REVIEW

File: patroni/postgresql.py [hunk 1]
```diff
@@ -389,6 +389,35 @@ def _initialize(self, config):
...
         return ret
 
+    def run_bootstrap_post_init(self, config):
+        """
+        runs a script after initdb is called and waits until completion.
+        passed: cluster name, parameters
+        """
+        if 'post_init' in config:
+            cmd = config['post_init']
+            r = self.get_local_address()
```
Comment (CyberDem0n) [inline patroni/postgresql.py]: I would like to replace lines 399-410 with following:
```python
r = self._local_connect_kwargs
if 'user' in r:
    connstring = 'postgres://{user}@{host}:{port}/{database}'.format(**r)
else:
    connstring = 'postgres://{host}:{port}/{database}'.format(**r)

env = self.write_pgpass(r) if 'password' in r else None
```

File: patroni/postgresql.py [hunk 2]
```diff
@@ -389,6 +389,35 @@ def _initialize(self, config):
...
+            if 'username' in self._superuser:
+                r.update({'user': self._superuser['username']})
+                connstring = 'postgres://{user}@{host}:{port}/{database}'.format(**r)
+            else:
+                connstring = 'postgres://{host}:{port}/{database}'.format(**r)
+            if 'password' in self._superuser:
+                r.update({'password': self._superuser['password']})
+                env = self.write_pgpass(r)
+            else:
+                env = os.environ.copy()
```
Comment (CyberDem0n) [inline patroni/postgresql.py]: post_init script doesn't need to know all environment of Patroni...

File: patroni/postgresql.py [hunk 3]
```diff
@@ -389,6 +389,35 @@ def _initialize(self, config):
...
+            try:
+                ret = subprocess.call(shlex.split(cmd) + [connstring], env=env)
+            except OSError:
+                logger.exception('post_init script %s failed', cmd)
+                return False
+            if ret != 0:
+                logger.exception('post_init script %s returned non-zero code %d', cmd, ret)
```
Comment (CyberDem0n) [inline patroni/postgresql.py]: I think logger.error would be enough in this case

File: patroni/postgresql.py (hunk 1 above)
Comment (zenitraM) [inline patroni/postgresql.py]: :+1: thanks! didn't notice local_connect_kwargs

File: docs/SETTINGS.rst [hunk 4]
```diff
@@ -32,6 +32,7 @@ Bootstrap configuration
         -  **options**: list of options for CREATE USER statement
             -  **- createrole**
             -  **- createdb**
+-  **post_init**: An additional script that will be executed after initializing the cluster. It will be passed a connection string URL to the cluster superuser as a parameter, as well as the location to the **pgpass* file on the `PGPASSENV` variable.
```
Comment (zenitraM) [inline docs/SETTINGS.rst]: Sounds good

**Comment by CyberDem0n** [comment]
:+1:

**Comment by alexeyklyukin** [comment]
👍

**Pull request merged**